modules/renter/uploadheap.go

```diff
@@ -236,6 +238,30 @@ func (r *Renter) managedBuildChunkHeap(hosts map[string]struct{}) {
 	// the heap.
 	id := r.mu.Lock()
 	for _, file := range r.files {
+		// check for local file
+		tf, exists := r.persist.Tracking[file.name]
+		if exists {
+			// Build maps that map contract id to its offline and goodForRenew
+			// status
+			contractIDs := make(map[types.FileContractID]struct{})
+			for cid := range file.contracts {
+				contractIDs[cid] = struct{}{}
+			}
+			goodForRenew := make(map[types.FileContractID]bool)
+			offline := make(map[types.FileContractID]bool)
+			for cid := range contractIDs {
+				resolvedID := r.hostContractor.ResolveID(cid)
+				cu, ok := r.hostContractor.ContractUtility(resolvedID)
+				goodForRenew[cid] = ok && cu.GoodForRenew
+				offline[cid] = r.hostContractor.IsOffline(resolvedID)
+			}
+			// Check if local file is missing and redundancy is less than 1
+			// log warning to renter log
+			if _, err := os.Stat(tf.RepairPath); os.IsNotExist(err) && file.redundancy(offline, goodForRenew) < 1 {
+				r.log.Println("File not found on disk:", tf.RepairPath)
```

Comment:
I'd probably change that message. It is not uncommon that a file is not found on disk but what we would like to warn the user about are files that are not found on disk and are possibly unrecoverable.